hummingbot/connector/connector/uniswap/uniswap_connector.py

```diff
@@ -0,0 +1,491 @@
+        return s_logger
+
+    def __init__(self,
+                 trading_pairs: List[str],
+                 wallet_private_key: str,
+                 ethereum_rpc_url: str,
+                 trading_required: bool = True
+                 ):
+        """
+        :param trading_pairs: a list of trading pairs
+        :param wallet_private_key: a private key for eth wallet
+        :param ethereum_rpc_url: this is usually infura RPC URL
+        :param trading_required: Whether actual trading is needed.
+        """
+        super().__init__()
+        self._trading_pairs = trading_pairs
+        tokens = set()
+        for trading_pair in trading_pairs:
+            tokens.update(set(trading_pair.split("-")))
+        self._token_addresses = get_erc20_token_addresses(tokens)
+        self._wallet_private_key = wallet_private_key
+        self._ethereum_rpc_url = ethereum_rpc_url
+        self._trading_required = trading_required
+        self._ev_loop = asyncio.get_event_loop()
+        self._shared_client = None
+        self._last_poll_timestamp = 0.0
+        self._in_flight_orders = {}
+        self._allowances = {}
+        self._status_polling_task = None
+        self._auto_approve_task = None
+        self._real_time_balance_update = False
+
+    @property
+    def name(self):
+        return "uniswap"
+
+    @property
+    def limit_orders(self) -> List[LimitOrder]:
+        return [
+            in_flight_order.to_limit_order()
+            for in_flight_order in self._in_flight_orders.values()
+        ]
+
+    async def auto_approve(self):
+        """
+        Automatically approves Uniswap contract as a spender for token in trading pairs.
+        It first checks if there are any already approved amount (allowance)
+        """
+        self.logger().info("Checking for allowances...")
+        self._allowances = await self.get_allowances()
+        for token, amount in self._allowances.items():
+            if amount <= s_decimal_0:
+                amount_approved = await self.approve_uniswap_spender(token)
+                if amount_approved > 0:
+                    self._allowances[token] = amount_approved
+                    await asyncio.sleep(2)
+                else:
+                    break
+
+    async def approve_uniswap_spender(self, token_symbol: str) -> Decimal:
+        """
+        Approves Uniswap contract as a spender for a token.
+        :param token_symbol: token to approve.
+        """
+        resp = await self._api_request("post",
+                                       "eth/approve",
+                                       {"tokenAddress": self._token_addresses[token_symbol],
+                                        "gasPrice": str(get_gas_price()),
+                                        # "decimals": self._token_decimals[token_symbol]  // if not supplied, gateway would treat it eth-like with 18 decimals
+                                        "connector": self.name})
```

Comment:
Since the approve API now requires connector name, can you please update `BalancerConnector` as well?